File: dehaze-python/app/models/entity/sys_ai_provider.py
from typing import Any

from sqlalchemy import BigInteger, Integer, SmallInteger, String
from sqlalchemy.dialects.mysql import JSON
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import BaseModel, SoftDeleteMixin


class SysAiProvider(BaseModel, SoftDeleteMixin):
    __tablename__ = "sys_ai_provider"
    __table_args__ = {"comment": "AI模型供应商配置表"}

    id: Mapped[int] = mapped_column(
        BigInteger, primary_key=True, autoincrement=True, comment="主键"
    )
    provider_code: Mapped[str] = mapped_column(
        String(32),
        unique=True,
        nullable=False,
        comment="供应商编码(openai;anthropic;deepseek;zhipu;qwen;custom)",
    )
    display_name: Mapped[str] = mapped_column(
        String(128), nullable=False, comment="显示名称(如OpenAI;Anthropic;DeepSeek)"
    )
    api_base_url: Mapped[str] = mapped_column(
        String(512), nullable=False, comment="API基础地址(如https://api.openai.com/v1)"
    )
    protocol_type: Mapped[str] = mapped_column(
        String(32),
        nullable=False,
        default="openai_compat",
        comment="协议类型(openai_compat:OpenAI兼容;anthropic:Claude原生)",
    )
    auth_type: Mapped[str] = mapped_column(
        String(32),
        nullable=False,
        default="bearer",
        comment="认证方式(bearer:Authorization Bearer;x-api-key:Anthropic风格;custom:自定义请求头)",
    )
    default_headers: Mapped[Any | None] = mapped_column(
        JSON, nullable=True, comment='默认请求头(JSON,如{"anthropic-version":"2023-06-01"})'
    )
    sort_order: Mapped[int] = mapped_column(
        Integer, nullable=False, default=0, comment="排序序号(数字越小越靠前)"
    )
    health_check_enabled: Mapped[int] = mapped_column(
        SmallInteger, nullable=False, default=1, comment="健康检查开关(1:开启,参与熔断判定;0:关闭)"
    )
    remark: Mapped[str | None] = mapped_column(
        String(512), nullable=True, comment="运维备注(账号归属/合同号/商务信息)"
    )
    status: Mapped[int] = mapped_column(
        SmallInteger, nullable=False, default=1, comment="状态(1:启用;0:禁用)"
    )

File: dehaze-go/internal/model/sys_ai_provider.go
package model

import "encoding/json"

// SysAiProvider AI模型供应商配置表
type SysAiProvider struct {
	BaseModel
	ProviderCode        string          `gorm:"column:provider_code;type:varchar(32);not null;comment:供应商编码" json:"providerCode"`
	DisplayName         string          `gorm:"column:display_name;type:varchar(128);not null;comment:显示名称" json:"displayName"`
	ApiBaseUrl          string          `gorm:"column:api_base_url;type:varchar(512);not null;comment:API基础地址" json:"apiBaseUrl"`
	ProtocolType        string          `gorm:"column:protocol_type;type:varchar(32);not null;default:openai_compat;comment:协议类型" json:"protocolType"`
	AuthType            string          `gorm:"column:auth_type;type:varchar(32);not null;default:bearer;comment:认证方式" json:"authType"`
	DefaultHeaders      json.RawMessage `gorm:"column:default_headers;type:json;comment:默认请求头" json:"defaultHeaders"`
	SortOrder           int             `gorm:"column:sort_order;type:int;not null;default:0;comment:排序序号" json:"sortOrder"`
	HealthCheckEnabled  int8            `gorm:"column:health_check_enabled;type:tinyint;not null;default:1;comment:健康检查开关" json:"healthCheckEnabled"`
	UserIdentityForward json.RawMessage `gorm:"column:user_identity_forward;type:json;comment:用户身份透传配置" json:"userIdentityForward"`
	Remark              *string         `gorm:"column:remark;type:varchar(512);comment:运维备注" json:"remark"`
	Status              int8            `gorm:"column:status;type:tinyint;not null;default:1;comment:状态(1:启用;0:禁用)" json:"status"`
	Deleted             int64           `gorm:"column:deleted;type:bigint;not null;default:0;comment:逻辑删除标识" json:"deleted"`
}

func (SysAiProvider) TableName() string {
	return "sys_ai_provider"
}
